src/main/scala/cqrs/domain/orders/Order.scala

```diff
@@ -42,19 +49,17 @@ class Order(maxOrderPrice: Double) extends PersistentActor with SettingsActor wi
       context become submitted
   }
 
-  def submitted: Receive = {
-    case msg ⇒
-      log.error("Order is completed. Will not process: {}", msg)
-      sender() ! Status.Failure(OrderIsSubmittedException(orderId))
-  }
-
-  override def receiveRecover: Receive = {
-    case event: Event ⇒
-      log.debug("Receiving recover message: {}", event)
-      updateState(event)
+  def uninitialized : Receive = {
+    case InitializeOrder ⇒
+      persist(OrderInitialized) { evt ⇒
+        log.debug(s"Order initialized {}", persistenceId)
+        updateState(evt)
+      }
+    case ReceiveTimeout ⇒ context stop self
```

Comment:
Passivate (http://doc.akka.io/docs/akka/2.3.9/contrib/cluster-sharding.html#Passivation) would be the right option here.

We can also safely passivate everywhere when a `ReceiveTimeout` is received.